frontend/components/branchViewLink.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { SITE } from '../propTypes';
+import appConfig from '../../config/app';
```

Comment:
Hmm, this is a little bit clunky. The `preview_hostname` variable in `config/app.js` is hardcoded to this:

```javascript
process.env.FEDERALIST_PREVIEW_HOSTNAME || 'http://localhost:1338'
```

That means that if `FEDERALIST_PREVIEW_HOSTNAME` is not set in the environment where this is built (in our case CI) then this won't work. That makes me a little nervous because that isn't really obvious.

Additionally, pulling configs into the frontend makes me nervous because secrets can land in those and then leak into the frontend code. With that in mind, my ask would be to add a comment to the app config saying as much.